zoia_lib/UI/ZOIALibrarian_main.py

```diff
@@ -1222,6 +1254,47 @@ def import_patch(self):
             self.msg.setStandardButtons(QMessageBox.Ok)
             self.msg.exec_()
 
+    def check_for_updates(self):
+        """Checks Github to see if there has been an app update."""
+
+        try:
+            url = 'https://github.com/meanmedianmoge/zoia_lib/releases/latest'
+            r = requests.get(url)
+            latest = r.url.split('/')[-1]
+        except Exception as e:
+            # Other unknown API connection error.
+            self.ui.statusbar.showMessage("Github connection failed.", timeout=5000)
+            self.msg.setWindowTitle("Github Error")
+            self.msg.setIcon(QMessageBox.Information)
+            self.msg.setText(
+                "Failed to retrieve the repo metadata from Github."
+            )
+            self.msg.setDetailedText(str(e))
+            self.msg.setStandardButtons(QMessageBox.Ok)
+            self.msg.exec_()
+            self.msg.setDetailedText(None)
+
+        curr_os = platform.system().lower()
+        choices = {"windows": "Windows", "darwin": "Mac", "linux": "Linux"}
+
+        if float(self._version) < float(latest):
```

Comment:
I like this check, but I think it may be worthwhile to display some form of information to the user if a new version isn't found. At the moment, the functionality in the app is that nothing happens if that's the case, which may lead users to believe the feature isn't working. 

This could even be as simple as displaying the info in the bottom status bar which is done with some other information.

Reply:
Good suggestion! I'll also edit it to reflect the split release for Macs.